notify/src/connection.rs

```diff
@@ -17,20 +17,32 @@ pub trait Connection: Clone + Debug + Send + Sync + 'static {
     fn is_closed(&self) -> bool;
 }
 
+#[derive(Clone, Debug)]
+pub enum ChannelType {
+    Closable,
+    Persistent,
+}
+
 #[derive(Clone, Debug)]
 pub struct ChannelConnection<N>
 where
     N: Notification,
 {
     sender: Sender<N>,
+    channel_type: ChannelType,
 }
 
 impl<N> ChannelConnection<N>
 where
     N: Notification,
 {
-    pub fn new(sender: Sender<N>) -> Self {
-        Self { sender }
+    pub fn new(sender: Sender<N>, channel_type: ChannelType) -> Self {
```

Comment:
since this seems like a temporary patch, I'd prefer that the usual `new` remains like before (defaulting to `closable`) and that we add a special `new_persistent` constructor

Reply:
This is exactly what we want to avoid.  This behavior has to be explicit as this hides the problem under the rug.  Any developer supplying a channel should expect the channel to remain alive for the lifetime the developer expects it to have.  Handing a channel to an external API and then having it closed is quite unexpected (at least in my experience).  If an API wants to control the lifetime of the channel, I would expect it to provide one. If I am providing one, I expect to be in control.

Reply:
ok